staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -52,21 +54,30 @@ type uniformClient struct {
 // expectPass indicates whether the QueueSet is expected to be fair.
 // expectedAllRequests indicates whether all requests are expected to get dispatched.
 func exerciseQueueSetUniformScenario(t *testing.T, name string, qs fq.QueueSet, sc uniformScenario,
-	evalDuration time.Duration, expectPass bool, expectedAllRequests bool,
+	evalDuration time.Duration,
+	expectPass, expectedAllRequests, expectInqueueMetrics, expectExecutingMetrics bool,
 	clk *clock.FakeEventClock, counter counter.GoRoutineCounter) {
 
 	now := time.Now()
 	t.Logf("%s: Start %s, clk=%p, grc=%p", clk.Now().Format(nsTimeFmt), name, clk, counter)
 	integrators := make([]test.Integrator, len(sc))
 	var failedCount uint64
+	expectedInqueue := ""
+	expectedExecuting := ""
+	if expectInqueueMetrics || expectExecutingMetrics {
+		metrics.Reset()
+	}
+	executions := make([]int32, len(sc))
 	for i, uc := range sc {
 		integrators[i] = test.NewIntegrator(clk)
+		fsName := fmt.Sprintf("client%d", i)
+		expectedInqueue = expectedInqueue + fmt.Sprintf(`				apiserver_flowcontrol_current_inqueue_requests{flowSchema=%q,priorityLevel=%q} 0%s`, fsName, name, "\n")
```

Comment:
From my reading of the code, timed out requests could get deducted from the wrong FS; I think a test should detect that if it happens (for any reason), but it's not clear that this one does?

E.g. after everything is done, the metrics should show zero executing requests, or the sum of rejected+processed needs to match what was sent, something like that?

Reply:
Yes, timeouts should be tested and this function does not do that.

Reply:
I have added some examination of metrics for rejection.